vshard/storage/init.lua

```diff
@@ -2945,18 +2943,22 @@ local function rebalancer_request_state()
         return nil, err
     end
     if not M.is_rebalancer_active or rebalancing_is_in_progress() then
-        return
+        return nil, lerror.make('Rebalancer is not active or is in progress')
     end
     local _bucket = box.space._bucket
     local status_index = _bucket.index.status
+    local repl_id = M.this_replica and M.this_replica.id or '_'
     if #status_index:select({BSENDING}, {limit = 1}) > 0 then
-        return
+        return nil, lerror.vshard(lerror.code.BUCKET_INVALID_STATE,
+                                  repl_id, 'sending')
     end
     if #status_index:select({BRECEIVING}, {limit = 1}) > 0 then
-        return
+        return nil, lerror.vshard(lerror.code.BUCKET_INVALID_STATE,
+                                  repl_id, 'receiving')
     end
     if #status_index:select({BGARBAGE}, {limit = 1}) > 0 then
-        return
+        return nil, lerror.vshard(lerror.code.BUCKET_INVALID_STATE,
+                                  repl_id, 'garbage')
     end
```

Comment:
The prefix `BUCKET_` suggests that this is a bucket-centric error. So like one specific bucket is bad or isn't found or alike. But this is actually a storage-centric or rebalancer-centric error. It signals that something can't be done because of the storage's state.

Also we already have here a generic code-less error above - `'Rebalancer is not active or is in progress'`.

I suggest not to invent anything new here and just return `lerror.make()` again with a formatted string. Anyway you never use this new error code for anything except tests.